src/v/kafka/client/partitioners.cc

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "kafka/client/partitioners.h"
+
+#include "hashing/murmur.h"
+
+#include <functional>
+#include <optional>
+#include <tuple>
+#include <type_traits>
+
+namespace kafka::client {
+
+namespace detail {
```

Comment:
@BenPope can you comment on the commit change if this is the same strategy that the java client uses, or if this is our own technique. We should probably do exactly what upstream uses as the default to keep compatibility.